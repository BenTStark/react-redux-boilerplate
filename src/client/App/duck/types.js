const LOGIN_REQUEST = "LOGIN_REQUEST";
const LOGIN_SUCCESS = "LOGIN_SUCCESS";
const LOGIN_ERROR = "LOGIN_ERROR";
const LOGOUT = "LOGOUT";
const TOGGLE_LOGIN_METHOD = "TOGGLE_LOGIN_METHOD";

export default {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_ERROR,
  LOGOUT,
  TOGGLE_LOGIN_METHOD
};
