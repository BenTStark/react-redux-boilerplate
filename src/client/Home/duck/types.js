const GET_OBJECT = "GET_OBJECT";
const TEST_ACTION = "TEST_ACTION";

export default {
  GET_OBJECT,
  TEST_ACTION
};
