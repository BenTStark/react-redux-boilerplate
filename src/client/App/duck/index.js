import authReducer from './reducers';
export default authReducer;
export { AppOperations } from './operations';
export { default as appTypes } from './types';
