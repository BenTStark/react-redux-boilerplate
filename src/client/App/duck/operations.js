// optional for API requests
import ActionCreators from "./actions";
import AuthService from "./auth.service";
import types from "./types";

const loginRequest = () => ActionCreators.loginRequest();
const loginSuccess = authInformation => ActionCreators.loginSuccess(authInformation);
const loginError = () => ActionCreators.loginError();
const logout = () => ActionCreators.logout();

const checkLogin = () => {
  return new Promise((resolve, reject) => {
    const payload = {},
      authInformation = {};
    authInformation.result = "";
    if (AuthService.loggedIn()) {
      payload.profile = AuthService.getProfile();
      payload.accessToken = AuthService.getAccessToken();
      authInformation.result = types.LOGIN_SUCCESS;
    }
    authInformation.payload = payload;
    resolve(authInformation);
  });
};

const authentication = () => {
  // Add callback for lock's `authenticated` event
  return new Promise((resolve, reject) => {
    const payload = {},
      authInformation = {};
    authInformation.result = "";
    authInformation.payload = payload;
    AuthService.lock.on("authenticated", authResult => {
      AuthService.lock.getUserInfo(authResult.accessToken, (error, profile) => {
        if (error) {
          authInformation.result = types.LOGIN_ERROR;
          reject(authInformation);
        } else {
          AuthService.setToken(authResult.idToken, authResult.accessToken); // static method
          AuthService.setProfile(profile); // static method
          payload.profile = profile;
          payload.accessToken = authResult.accessToken;
          authInformation.result = types.LOGIN_SUCCESS;
          authInformation.payload = payload;
          AuthService.lock.hide();
          resolve(authInformation);
        }
      });
    });
    // Add callback for lock's `authorization_error` event
    AuthService.lock.on("authorization_error", error => {
      authInformation.result = types.LOGIN_ERROR;
      reject(authInformation);
    });
  });
};

const handleLogin = () => {
  AuthService.login();
};

const handleLogout = () => {
  AuthService.logout();
};

export const AppOperations = {
  handleLogin,
  loginRequest,
  loginSuccess,
  loginError,
  handleLogout,
  logout,
  checkLogin,
  authentication
};

//export default AppOperations;
