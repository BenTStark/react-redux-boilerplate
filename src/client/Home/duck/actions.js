import types from "./types.js";

const defaultAction = value => {
  return {
    type: types.DEFAULT_ACTION,
    value: value
  };
};

export default {
  defaultAction
};
