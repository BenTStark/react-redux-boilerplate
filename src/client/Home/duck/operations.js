import ActionCreators from "./actions";

// URLs for API us are build here, but used in container!
const getDataUrl = "https://mirrorpi.ddns.net/api";

const getObj = objAsync => ActionCreators.getObj(objAsync);

const testAction = () => ActionCreators.testAction();

export const HomeOperations = {
  getObj,
  getDataUrl,
  testAction
};
