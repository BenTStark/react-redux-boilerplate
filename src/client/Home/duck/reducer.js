import types from "./types";

const initalState = {
  payload: null,
  test: false
};

function getObj(state, getObjAsnyc) {
  return {
    ...state,
    payload: getObjAsnyc
  };
}

function testAction(state) {
  return {
    ...state,
    test: !state.test
  };
}

const homeReducer = (state = initalState, action) => {
  switch (action.type) {
    case types.GET_OBJECT:
      return getObj(state, action.payload);
    case types.TEST_ACTION:
      return testAction(state);
    default:
      return state;
  }
};

export default homeReducer;
