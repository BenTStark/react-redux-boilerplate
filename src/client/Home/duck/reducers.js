import types from "./types";

const initalState = {
  payload: null,
  test: false
};

function getObj(state, getObjAsnyc) {
  return {
    ...state,
    payload: getObjAsnyc
  };
}

function testAction(state) {
  return {
    ...state,
    test: !state.test
  };
}

const homeReducer = (state = initalState, action) => {
  console.log("reducer");
  console.log(action);
  switch (action.type) {
    case types.GET_OBJECT:
      return getObj(state, action.payload);
    case types.TEST_ACTIONS:
      return testAction(state);
    default:
      return state;
  }
};

export default homeReducer;
