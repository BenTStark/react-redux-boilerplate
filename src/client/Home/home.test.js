import HomeComponent from "./home.component";
import HomeContainer from "./home.container";
import { HomeOperations } from "./duck/operations";
import types from "./duck/types";

import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";

// Snapshot for Home React Component
describe(">>> HomeComponent - Snapshot", () => {
  it("+++capturing Snapshot of HomeComponent", () => {
    const renderedValue = shallow(
      <HomeComponent auth={{}} home={{}} />
    );
    expect(toJson(renderedValue)).toMatchSnapshot();
  });
});

// Component Test
describe(">>> HomeComponent --- Shallow Render React Components", () => {
  const wrapper = shallow(<HomeComponent auth={{}} home={{}} />);

  it("+++ render the COMPONENT", () => {
    expect(wrapper.length).toEqual(1);
  });
});

// Container Test
describe(">>> HomeContainer --- REACT-REDUX (Shallow + passing the {store} directly)", () => {
  const initialState = {
    authReducer: {
      profile: null,
      accessToken: null,
      loginRequest: false,
      loginSuccess: false,
      loginError: false
    },
    homeReducer: {
      payload: null,
      test: false
    }
  };
  const mockStore = configureStore();
  let store, container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<HomeContainer store={store} />);
  });

  it("+++ check Prop matches with initialState", () => {
    expect(container.prop("home")).toEqual(initialState.homeReducer);
    expect(container.prop("auth")).toEqual(initialState.authReducer);
  });

  describe(">>> Test Axios Request", () => {
    const mockData = { response: "content" };
    beforeEach(() => {
      const mock = new MockAdapter(axios);
      mock.onGet(HomeOperations.getDataUrl).reply(200, mockData);
      container
        .dive()
        .find("#axios")
        .simulate("click");
    });

    it("+++ Click Request Button", () => {
      expect(store.getActions()).toEqual([
        {
          type: types.GET_OBJECT,
          payload: mockData
        }
      ]);
    });
  });
});
