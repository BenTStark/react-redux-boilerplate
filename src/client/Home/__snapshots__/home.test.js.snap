// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> HomeComponent - Snapshot +++capturing Snapshot of HomeComponent 1`] = `
<div>
  <p>
    Auth State
  </p>
  <pre>
    {}
  </pre>
  <p>
    Home State
  </p>
  <button
    onClick={[Function]}
  >
    Test Action
  </button>
  <button
    id="axios"
    onClick={[Function]}
  >
    Test Axios
  </button>
  <pre>
    {}
  </pre>
  <p>
    Local Storage
  </p>
  <pre>
    {}
  </pre>
</div>
`;
